Base/Logic/vtkSlicerApplicationLogic.cxx

```diff
@@ -98,6 +98,15 @@ vtkSlicerApplicationLogic::~vtkSlicerApplicationLogic()
     this->ProcessingThread.join();
   }
 
+  for (auto& thread : this->NetworkingThreads)
+  {
+    if (thread.joinable())
+    {
+      thread.join();
+    }
+  }
+  this->NetworkingThreads.clear();
+
```

Comment:
```suggestion
  this->TerminateProcessingThread();
```

I think should also move the code above into `TerminateProcessingThread`

Reply:
For reference the original code in `TerminateProcessingThread()` and the one in the destructor `~vtkSlicerApplicationLogic` were both introduced in the same commit aef3c715c5 (`BUG: CommandLineModules need to place a request to the main thread to reload data and display results. ENH: Moved processing threading control to vtkSlicerApplicationLogic. Moved most of the interthread communication to vtkSlicerApplicationLogic.  Interthread communication for Warnings/Errors/Information still in vtkSlicerApplication`, 2006-12-12)

Reply:
I've updated the code to use `this->TerminateProcessingThread();`, please review and approve if looks good.